### PR TITLE
[MM-38615] Permalink Previews for DMs/GMs

### PR DESCRIPTION
**Pull request by mylonsuren**
#### Summary

Adds permalink previews for DMs and GMs.

#### Ticket Link

Fixes [MM-38615](https://mattermost.atlassian.net/browse/MM-38615)

#### Related Pull Requests

- Has server changes https://github.com/mattermost/mattermost-server/pull/19765

#### Release Note

```release-note
Adds permalink previews for DMs/GMs
```

### CODE REVIEW

**Comment by mylonsuren** [comment]
Need to fix a bug with the DM channel name not appearing. Will ping when this PR is ready for review again.

**Comment by mylonsuren** [comment]
@mkraft @BenCookie95 Both PRs should be good for review now.

File: components/post_view/post_message_preview/index.ts [hunk 1]
```diff
@@ -30,8 +34,10 @@ type Props = {
 
 function mapStateToProps(state: GlobalState, ownProps: Props) {
```
Comment (BenCookie95) [inline components/post_view/post_message_preview/index.ts:35]: Turn this into a makeMapStateToProps and move `makeGetChannel`. See here for an example, /components/threading/global_threads/thread_item/index.ts

**Review by BenCookie95 — COMMENTED** [review]
Nice work, just one small comment!

File: components/post_view/post_message_preview/index.ts (hunk 1 above)
Comment (mylonsuren) [reply components/post_view/post_message_preview/index.ts]: Thanks, I've made the changes 👍

**Review by BenCookie95 — APPROVED** [review]

**Review by mkraft — APPROVED** [review]

**Review by furqanmlk — APPROVED** [review]
GM permalink is rendering fine in GM ✅ 
GM permalink is rendering fine in DM ✅ 
GM permalink is rendering fine in Private Channel ✅ 
GM permalink is rendering fine in Public Channel ✅ 
DM permalink is rendering fine in GM ✅ 
DM permalink is rendering fine in DM ✅ 
DM permalink is rendering fine in Private Channel ✅ 
DM permalink is rendering fine in Public Channel ✅

**Pull request merged**